structures/README.adoc

```diff
@@ -0,0 +1,54 @@
+
+Rationale::
+especially when writing automation in a team, it is important to have a certain level of consistence and make sure everybody has the same understanding.
+By lack of doing so, your automation becomes unreadable and difficult to grasp for new members or even for existing members.
++
+This structure will also help you to have a consistent level of modelization so that re-usability becomes easier.
+Metaphorically speaking, only if stones have been cut at roughly the same size, can they be properly used to build a house.
+
+Examples::
+The following is only one example of how to structure your content but has proven robust enough on multiple occasions.
++
+.Structure of Automation
+image::ansible_structures.svg[a hierarchy of landscape type function and component]
++
+* a _landscape_ is anything you want to deploy at once, the underlay of your cloud, a three tiers application, a complete application cluster...
+  This level is represented at best by a Tower/AWX workflow, potentially by a "playbook of playbooks", i.e. a playbook made of imported _type_ playbooks.
+* a _type_ must be defined such that each managed host has one and only one type, applicable using a unique playbook.
+* each type is then made of multiple _functions_, represented by roles, so that the same function used by the same _type_ can be re-used, written only once.
+* and finally _components_ are used to split a _function_ in maintainable bits. By default a component is a task file within the _function_-role, if the role becomes too big, there is a case for splitting the _function_ role into multiple _component_ roles.
++
+NOTE: if _functions_ are defined mostly for re-usability purposes, _components_ are more used for maintainability / readability purposes. A re-usable component might be a candidate for promotion to a function.
++
+Let's have a more concrete example to clarify:
++
+* as already written, a _landscape_ could be a three tier application with web-front-end, middleware and database.
+  We would probably create a workflow to deploy this landscape at once.
+* our types would be relatively obvious here as we would have "web-front-end server", "middleware server" and "database server".
+  Each type can be fully deployed by one and only one playbook (avoid having numbered playbooks and instructions on how to call them one after the other).
+* each server type is then made of one or more _functions_, each implemented as a role.
+  For example, the middleware server type could be made of a "virtual machine" (to create the virtual machine hosting the middleware server), a "base Linux OS" and a "JBoss application server" function.
+* and then the base OS role could be made of multiple components (DNS, NTP, SSH, etc), each represented by a separate `tasks/{component}.yml` file, included or imported from the `tasks/main.yml` file of the __function__-role.
```

Comment:
```suggestion
* and then the base OS role could be made of multiple components (DNS, NTP, SSH, etc), each represented by a separate `tasks/{component}.yml` file, included or imported from the `tasks/main.yml` file of the _function_-role.
```

Reply:
Yeah, it should be consistent, Vim renders it better though. Anyway, done, consistency is more important.